delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/SchemaClaimSuite.scala

```diff
@@ -0,0 +1,70 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.resources
+
+import cats.effect.IO
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.nxv
+import ch.epfl.bluebrain.nexus.delta.rdf.shacl.ValidationReport
+import ch.epfl.bluebrain.nexus.delta.sdk.identities.model.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.resources.SchemaClaim.SubmitOnDefinedSchema
+import ch.epfl.bluebrain.nexus.delta.sdk.resources.ValidationResult._
+import ch.epfl.bluebrain.nexus.delta.sdk.resources.model.ResourceRejection.SchemaIsMandatory
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity.User
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{Label, ProjectRef, ResourceRef}
+import ch.epfl.bluebrain.nexus.testkit.mu.NexusSuite
+import io.circe.Json
+import munit.Location
+
+class SchemaClaimSuite extends NexusSuite {
+
+  private val project       = ProjectRef.unsafe("org", "proj")
+  private val definedSchema = ResourceRef.Revision(nxv + "schema", 1)
+  private val unconstrained = ResourceRef.Revision(Vocabulary.schemas.resources, 1)
+
+  private val subject = User("myuser", Label.unsafe("myrealm"))
+  private val caller  = Caller(subject, Set.empty)
+
+  private val noValidation = NoValidation(project)
+  private val validated    = Validated(project, definedSchema, ValidationReport.unsafe(conforms = true, 42, Json.Null))
+
+  private val schemaIsMandatory = SchemaIsMandatory(project)
+
+  private def submitOnDefinedSchema: SubmitOnDefinedSchema = (_, _, _) => IO.pure(validated)
+
+  private def assertValidation(claim: SchemaClaim)(implicit location: Location) =
+    claim.validate(enforceSchema = true)(submitOnDefinedSchema).assertEquals(validated) >>
+      claim.validate(enforceSchema = false)(submitOnDefinedSchema).assertEquals(validated)
+
+  private def assetNoValidation(claim: SchemaClaim)(implicit location: Location) =
+    claim.validate(enforceSchema = true)(submitOnDefinedSchema).assertEquals(noValidation) >>
+      claim.validate(enforceSchema = true)(submitOnDefinedSchema).assertEquals(noValidation)
+
+  private def failOrNoValidation(claim: SchemaClaim)(implicit location: Location) =
+    claim.validate(enforceSchema = true)(submitOnDefinedSchema).interceptEquals(schemaIsMandatory) >>
+      claim.validate(enforceSchema = false)(submitOnDefinedSchema).assertEquals(noValidation)
+
+  test("Create with a schema runs validation") {
+    val claim = SchemaClaim.onCreate(project, definedSchema, caller)
+    assertValidation(claim)
+  }
+
+  test("Create unconstrained fails or skip validation") {
+    val claim = SchemaClaim.onCreate(project, unconstrained, caller)
+    failOrNoValidation(claim)
+  }
```

Comment:
I would consider changing this test to make it clear what the conditions are. Really, this should be two tests. It's better that the test is more verbose but more understandable